src/TableOfContents/index.tsx

```diff
@@ -0,0 +1,226 @@
+import { Button, Drawer, Icon, IconName } from '@blueprintjs/core';
+import cn from 'classnames';
+import * as React from 'react';
+import { useIsMobile } from '../_hooks/useIsMobile';
+import { ComponentsContext } from '../Provider';
+import { ScrollContainer } from '../ScrollContainer';
+import { IContentsNode } from '../types';
+import { deserializeSrn } from '../utils';
+
+export interface ITableOfContents {
+  // Precomputed list of nodes instead of items. This is useful if you want to customize the ordering of the tree.
+  contents: IContentsNode[];
+
+  // SRN of the active node
+  srn?: string;
```

Comment:
I don't think we need "srn" anymore since `IContentsNode` has an `isActive` property

```suggestion
```